File: week_1/day-3-two-pointer/remove_duplicate_follow_up_questions/remove_duplicates_allow_twice.js
let nums = [1, 1, 1, 2, 2, 3];

function removeDuplicatesTwice(nums) {
    if (nums.length <= 2) {
        return nums.length;
    }

    let slow = 2;

    for (let fast = 2; fast < nums.length; fast++) {
        if (nums[fast] !== nums[slow - 2]) {
            nums[slow] = nums[fast];
            slow++;
        }
    }

    return slow;
}

let length = removeDuplicatesTwice(nums);
console.log(nums.slice(0, length)); // Output: [1, 1, 2, 2, 3]

File: week_1/day-3-two-pointer/remove_duplicate_follow_up_questions/remove_duplicates_allow_twice.py
nums = [1,1,1,2,2,3]

def removeDuplicatesTwice(nums):
    if len(nums) <= 2:
        return len(nums)

    slow = 2
    for fast in range(2, len(nums)):
        if nums[fast] != nums[slow - 2]:
            nums[slow] = nums[fast]
            slow += 1

    return slow

length = removeDuplicatesTwice(nums)
print(nums[:length])  # Output: [1, 1, 2, 2, 3]
